plugins/modules/aws_msk_cluster.py

```diff
@@ -0,0 +1,827 @@
+#!/usr/bin/python
+# Copyright: (c) 2021, Daniil Kupchenko (@oukooveu)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: aws_msk_cluster
+short_description: Manage Amazon MSK clusters.
+version_added: "2.0.0"
+requirements:
+    - botocore >= 1.17.42
+    - boto3 >= 1.17.9
+description:
+    - Create, delete and modify Amazon MSK (Managed Streaming for Apache Kafka) clusters.
+author:
+    - Daniil Kupchenko (@oukooveu)
+options:
+    state:
+        description: Create (present) or delete (absent) cluster.
+        choices: ['present', 'absent']
+        type: str
+        default: 'present'
+    name:
+        description: The name of the cluster.
+        required: true
+        type: str
+    version:
+        description:
+            - The version of Apache Kafka.
+            - This version should exist in given configuration.
+            - This parameter is required when I(state=present).
+        type: str
+    configuration_arn:
+        description:
+            - ARN of the configuration to use.
+            - This parameter is required when I(state=present).
+        type: str
+    configuration_revision:
+        description:
+            - The revision of the configuration to use.
+            - This parameter is required when I(state=present).
+        type: int
+    nodes:
+        description: The number of broker nodes in the cluster. Should be greater or equal to two.
+        type: int
+        default: 3
+    instance_type:
+        description:
+            - The type of Amazon EC2 instances to use for Kafka brokers.
+            - Update operation requires boto3 version >= 1.16.58
+        choices:
+            - kafka.t3.small
+            - kafka.m5.large
+            - kafka.m5.xlarge
+            - kafka.m5.2xlarge
+            - kafka.m5.4xlarge
+        default: kafka.t3.small
+        type: str
+    ebs_volume_size:
+        description: The size in GiB of the EBS volume for the data drive on each broker node.
+        type: int
+        default: 100
+    subnets:
+        description:
+            - The list of subnets to connect to in the client virtual private cloud (VPC).
+              AWS creates elastic network interfaces inside these subnets. Client applications use
+              elastic network interfaces to produce and consume data.
+              Client subnets can't be in Availability Zone us-east-1e.
```

Comment:
```suggestion
            - Client subnets can't be in Availability Zone us-east-1e.
```
For any other reviewers, this is an AWS restriction and they only list us-east-1e
https://docs.amazonaws.cn/en_us/msk/1.0/apireference/clusters.html